src/Backups/BackupCoordinationFileInfos.cpp

```diff
@@ -194,7 +194,7 @@ void BackupCoordinationFileInfos::prepare() const
                 if (inserted)
                 {
                     /// Found a new file.
-                    info.data_file_name = info.file_name;
+                    info.data_file_name = keys_gen->generate(info.file_name, false, {}).serialize();
```

Comment:
If backup is not plain then every file with unique checksum is stored only once. So I think we could just convert `info.checksum` to String, maybe split it into two parts, and use it as `info.data_file_name` here. I suppose it would also be useful to keep the internal structure of backups simpler if we avoid introducing new random identifiers here.